### PR TITLE
Report for DEIMOS PD-8 and other docs

### PR DESCRIPTION
**Pull request by deborape**
As titled, I added a report for DEIMOS PD-8 requirement: "As a user, I expect products that are associated with my mask definition (object names, object positions)".

I wrote a unit test for DEIMOS overlapping alignment slits and I updated an existing report. This shows that PypeIt meets also PD-42 requirement: "Deal with overlapping alignment boxes that exceed 4” ". 

I have also updates some docs.

### CODE REVIEW

**Review by profxj — APPROVED** [review]
Wow, this is great!
Thanks!

File: doc/construct_template.rst [hunk 1]
```diff
@@ -46,11 +49,12 @@ wavelength ranges, and master wavecalib files into a final template.
...
-     shift_wave=True)
+.. code-block:: python
+
+    from pypeit.core.wavecal import templates
+
+    templates.build_template(wfiles, slits, wv_cuts, binspec, outroot, ifiles=ifiles, det_cut=det_cut, chk=True,
+                             normalize=False, lowredux=False, subtract_conti=True, overwrite=overwrite, shift_wave=True)
 
-This produces a file called 'outroot' that contains the template. It also
+This produces a file called `outroot` that contains the template. It also
```
Comment (kbwestfall) [inline doc/construct_template.rst:59]: I mix this up, as well, but I think if you want  outroot to appear like `outroot`, you need to use two quote characters; i.e.,
```
``outroot``
```

It's a difference in Markdown and restructuredText syntax.  As it is, I think this only puts outroot in italics.

File: doc/dev/deimos_radec_object.rst [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+and :ref:`master_slits` for a description of the provided information and for a way to visualize them).
+
+Then, :func:`~pypeit.slittrace.SlitTraceSet.assign_maskinfo` measures the distance in pixels (converted
+then in arcsec) between the traced object and the center of the slit and, using the coordinates of the
+slit center available from the slitmask design, estimates the coordinates of the extracted spectrum.
+
+``PypeIt`` goes through all the slits in a selected detector and for each extracted spectrum checks if
+the measured coordinates are within a certain tolerance (see `Application`_ for details on how to control the
+value of this parameter) of the expected coordinates for the targeted object. If the answer is yes, the 1D 
+spectrum `RA`, `DEC` and `MASKDEF_OBJNAME` are updated with the coordinates and name of the
```
Comment (kbwestfall) [inline doc/dev/deimos_radec_object.rst:42]: Same thing here about the single vs. double quote characters.

**Review by kbwestfall — APPROVED** [review]
Looks great!  Just a couple nuisance comments.

File: doc/construct_template.rst (hunk 1 above)
Comment (deborape) [reply doc/construct_template.rst:59]: Thanks! Done.

File: doc/dev/deimos_radec_object.rst (hunk 2 above)
Comment (deborape) [reply doc/dev/deimos_radec_object.rst:42]: Thanks.

**Comment by deborape** [comment]
Ok. I'm merging.

**Pull request merged**